### DISCUSSION TITLE
Is it possibile to order and paginate the TimeSeriesEntry descending when queryed?

### DISCUSSION THREAD

**Discussion by ynnova** [ravendb/ravendb#19111] [General]
I need to build a page where to filter, analyze and order a TimeSerie. In order to do that I need (also) to paginate and order the entries descending (the most recent upper).

Is it possible natively?

**Comment by ayende** [comment]
You can limit the range in time that you'll get, and you can set the grouping to a known value.
So you can say: Give me all results for the past 48 hours, grouped by hours (giving you 48 items).

But you can't limit that by number of results.

What exactly is it that you are trying to do?

Something that you can do, however, is to limit the range in _time_.

So you can do something like:

```
from 'Employees'
where id() == 'employees/7-A'
select timeseries(from 'HeartRates' last '2 minutes' group by 1s select avg()  )
```

That gives you 3 results in the range, then you can use the `between` to go back in ranges.

**Reply by ynnova** [reply]
Thank you Oren,

what i want to achieve is to paginate through the timeseries data (and this is what I do in the following snippet.

Also: I would like to order the paginated data by the most recent first.

I will copy/paste part of my code, so you can better understand what i'm trying to do:

```
var t1 = RavenSession
                    .TimeSeriesFor(request.DcaId, "_b0")
                    .GetAsync(DateTime.MinValue, DateTime.MaxValue, request.Query.Skip, request.Query.Take);

                var t2 = RavenSession.Advanced.DocumentStore.Operations
                    .ForDatabase(TenantDescriptor.DatabaseName)
                    .SendAsync(new GetTimeSeriesStatisticsOperation(request.DcaId));

                Task.WaitAll(t1, t2);
                
                var tsee = t1.Result;
                var stats = t2.Result;

                var b0Stats = stats.TimeSeries.SingleOrDefault(m => m.Name == "_b0");
                
// code deleted to simplify

                reply.Result.TotalResults = b0Stats?.NumberOfEntries ?? reply.Items.Count;

```

**Reply by shiranshalom** [reply]
Hi @ynnova,
If I understand correctly, you’re looking for a way to retrieve all time series entries in descending order.

Here’s how you can achieve that:
1. Define a New Index for the relevant time series. For example, you can create an index like this:
```
private class MyTimeSeriesIndex : AbstractTimeSeriesIndexCreationTask<Company>
{
	public class IndexEntry
	{
		public double Value;
		public DateTime Timestamp;
		public string Tag;
	}

	public MyTimeSeriesIndex()
	{
		AddMap("heartrate", timeSeries => from ts in timeSeries
									  from entry in ts.Entries
									  select new IndexEntry
									  {
										  Value = entry.Value,
										  Timestamp = entry.Timestamp,
										  Tag = entry.Tag
									  });
	}
}
```
2. Query the Index and order by descending:
```
using (var session = store.OpenSession())
{
	var q = session.Query<MyTimeSeriesIndex.IndexEntry, MyTimeSeriesIndex>()
		.OrderByDescending(x => x.Timestamp)
		.Skip(10)
		.Take(8)
		.ToList();
}
```

Hope this answers your question. If you have any further queries or need additional assistance, please feel free to ask!
